sound/soc/sof/intel/hda-dai.c

```diff
@@ -442,6 +442,11 @@ static const struct snd_soc_dai_ops hda_link_dai_ops = {
 
 #endif
 
+/* the SSP dma_data is only using a flag for now */
+struct ssp_dai_dma_data {
+	bool suspended;
```

Comment:
I would have used a flag for opposite meaning, `dai_ready`.
In `.prepare` then we can have
```
if (!dma_data->dai_ready) {
	ret = ssp_dai_setup_or_free(substream, dai, true);
	if (!ret)
		dma_data->dai_ready = true;
}
```
In other `.trigger` we swap the flag to false, from `.hw_params` it is going to come with false from the kzalloc().

Reply:
'ready' means nothing to me. the intent is that we track the 'suspended' case only. that's the only thing we care about.

Reply:
I see. It would make the code simpler if the state of the DAI is tracked, which would cover the case when the DAI is suspended. When it is `suspended`, it is _not_ `set_up`, iow the DAI is _not_ `ready`.

I would be tempted to note somewhere that we use the otherwise unused playback/capture_dma_data for local purpose, that other components in the setup does not use the dma_data.

Reply:
> @plbossart, this is partly why I was suggesting to track the `set_up` / `free` state of the DAI instead of a the `suspended` state only.

those are different things. the suspended case makes sure we deal with the suspend case. I added a configured flag in the last update to keep track of hw_params/hw_free potential issues.